### DISCUSSION TITLE
Welcome to MockMe Discussions!

### DISCUSSION THREAD

**Discussion by connorivy** [connorivy/MockMe#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by marchewek** [comment]
I've always felt that .NET is limited when it comes to following good practices.

One notable example would be the Dependency Injection support that took MS ages to implement. 

Another one would be the lack of mocking mechanisms provided by MS; save for MS Fakes that 1. were available only in the most expensive VS subscription, 2. had very unintuitive API and 3. running them in an automated infrastructure to this day requires an installed VS. Someone would always counter the need to mock concrete classes by a workaround: "add an interface" or "make the members virtual". No. These should be very conscious design decisions (the language constructs are not meant to be used mindlessly) and if they weren't employed in the first place, then consequently they shouldn't be incorporated solely for enabling the testing. Alas, they were due to the lack of alternatives. For years we were limited to: extremely expensive tools like MS good-luck-with-that Fakes, JustMock or TypeMock and libraries with poor developer experience like pose or AutoFake.

This library is something else. Great job! 💪 Finally there is something that looks promising 😄

I'm keeping my fingers crossed for this library to gain more traction 😃

**Reply by connorivy** [reply]
@marchewek Thanks for the encouraging comment! I completely agree with you about adding interfaces and virtual members. Please let me know if you have any issues or suggestions as you use the library.
